packages/neuron-ui/src/components/WalletEditor/index.tsx

```diff
@@ -1,32 +1,48 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { RouteComponentProps } from 'react-router-dom'
-import { Card, Form, Button, Alert } from 'react-bootstrap'
+import { Card, Form, Button } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
 
-import InputWalletPasswordDialog, { CheckType } from '../Settings/InputWalletPasswordDialog'
 import { ContentProps } from '../../containers/MainContent'
 import InlineInput, { InputProps } from '../../widgets/InlineInput'
 import { MainActions } from '../../containers/MainContent/reducer'
-import { Wallet } from '../../contexts/NeuronWallet'
+import { useNeuronWallet } from '../../utils/hooks'
+import InputWalletPasswordDialog, { CheckType } from '../Settings/InputWalletPasswordDialog'
 import Dialog from '../../widgets/Dialog'
+import { Routes } from '../../utils/const'
 
-export default (props: React.PropsWithoutRef<ContentProps & RouteComponentProps<{ wallet: string }>>) => {
+export default (props: React.PropsWithoutRef<ContentProps & RouteComponentProps<{ id: string }>>) => {
   const { match, dialog, dispatch } = props
   const { params } = match
   const [t] = useTranslation()
+  const {
+    settings: { wallets },
+    messages: errorMessage,
+  } = useNeuronWallet()
 
-  const myWallet: Wallet = JSON.parse(params.wallet)
-  const [walletName, setWalletName] = useState(myWallet.name)
+  const myWallet = wallets.find(wallet => wallet.id === params.id)
+  const [walletName, setWalletName] = useState('')
   const [password, setPassword] = useState('')
   const [confirmPassword, setConfirmPassword] = useState('')
-  const [errorMsg, setErrorMsg] = useState('')
+
+  useEffect(() => {
```

Comment:
Maybe the dialog is unnecessary. 

Input password with other fields versus a dialog for the password, which one is better?

Reply:
Many wallet would only ask and check password when user clicks 'save', 'send', etc.

Reply:
Also think of this: many devices including macOS computers now can handle password with Touch ID or Face ID. For this, we don't even have a password field for the form, but we'll usually trigger a confirm dialog.